Update free Lesson patterns.

aaronfc opened the pull request:
Fixes partially #5210

### Changes proposed in this Pull Request
* Update existing lesson patterns.
* Add: default lesson.
* Add: default lesson with Quiz.
* Add: Zoom/Meeting

### Testing instructions
- Create a new Lesson.
- Check that all the patterns appear and look as they should.

### Screenshot / Video
Screenshots using Divi theme.

#### List of Lesson Patterns (only free ones)
![image](https://user-images.githubusercontent.com/799065/172837428-c2d40406-4bad-4e8d-8efa-4ad4cfb02a1c.png)

#### Default (*new)
![image](https://user-images.githubusercontent.com/799065/172837560-b911b7fd-d5ff-4380-bd65-fcd16ded7d26.png)

#### Default with Quiz (*new)
![image](https://user-images.githubusercontent.com/799065/172837724-605372ff-b12a-4774-9c20-04aec0ff158d.png)

#### Video Lesson
![image](https://user-images.githubusercontent.com/799065/172837867-33c6abcb-f7ae-4c2a-a394-abd9f7cb2bf5.png)

#### Discussion Question Lesson
![image](https://user-images.githubusercontent.com/799065/172837973-473d9eeb-f037-4187-b0a7-ca5c58312797.png)

#### Files to Download
![image](https://user-images.githubusercontent.com/799065/172838207-23fd697a-6452-4a75-a691-bcd1059f9baa.png)

#### Zoom/Meeting (*new)
![image](https://user-images.githubusercontent.com/799065/172838611-4d910828-d76e-4a7c-bfe7-b788180438cd.png)

### Known bugs
- [x] Lesson actions should be at the bottom also for the Default with Quiz pattern. FIXED! ✅ 
- [ ] The default **teamplate** for the lessons is different from the Default pattern. @renatho is working on a solution to reuse the pattern in the template for the Courses. We will reuse the same for the Lessons.

aaronfc committed:
Add Quiz to Video lesson pattern.

aaronfc committed:
Add Quiz to Discussion Question lesson pattern.

aaronfc committed:
Update Files to Download pattern.

- Include Quiz.
- Add heading.

aaronfc committed:
Add Default lesson pattern.

aaronfc committed:
Add Default with Quiz lesson pattern.

aaronfc committed:
Add Zoom/Meeting lesson pattern.

aaronfc committed:
Include new patterns in patterns list.

aaronfc committed:
Remove Quiz block from Discussions, Files and Video lesson patterns.

aaronfc committed:
Escape output from wp_date call.

aaronfc committed:
Move lesson actions block to bottom for Default with Quiz pattern.

alexsanford commented:
I am not a designer, but these patterns still don't look really compelling to me. When I look at the designs from the post (p6rkRX-3zM-p2), it still doesn't seem to be translating to the actual modal.

A few things I think are contributing:

- The design post doesn't include the "Lesson Action" buttons. They are adding clutter to the previews.
- The design post doesn't deal with the "Default Lesson" patterns, which look very empty.
- The design post uses large, bold, visible headings, but our patterns do not.
- The design post has some "Testimonial" patterns at the top, with nice images, which gives some grounding for the overall design. But I don't think we're actually going to implement these patterns for lessons, right?
- There are other differences between the look of the designs in the post, and the actual look of the blocks. For example, the flashcard block in the design does not match the look of the actual flashcard block (in the [premium patterns PR](https://github.com/Automattic/sensei-pro/pull/1271)); the text is more bold, the "Flip card" link has an icon.

cc @Luchadores @burtrw

renatho commented:
I didn't test it running, but the code looks good to me!

I'd just ask if you could base it on [this branch](https://github.com/Automattic/sensei/pull/5246), and update the lesson CPT to use the default pattern, so we avoid the duplicated code to maintain. 😀

burtrw commented:
A note that Alex and I are playing with a solution to force a common aspect ratio of all thumbnails, center the content if it fits (if it is too long, then truncate the bottom), and add a little more padding between each pattern. This will help clean up the experience and make it easier to distinguish between each pattern.

alexsanford commented:
Here's what it looks like after cherry picking the CSS change from https://github.com/Automattic/sensei/pull/5251

<img width="822" alt="Screen Shot 2022-06-09 at 17 10 36" src="https://user-images.githubusercontent.com/842193/172935471-4f559bb4-4f26-47d9-9f70-5a9fb55c1a94.png">

cc @burtrw

aaronfc commented:
@alexsanford good points on the design and cheers on your change for consistent preview sizes, it looks so much better with that.

It is true that the designs are different to the original ones. Maybe I misunderstood the task because I didn't even check them. I just assumed I would take whatever @burtrw provided in the demo site as is.
I am adding my input to some of your points: 

> The design post doesn't include the "Lesson Action" buttons. They are adding clutter to the previews.

This buttons were included because they are needed in case you are not using Learning Mode. We discussed about adding/removing depending on whether you had LM on or not, but finally decided on adding it always at the bottom.

> The design post doesn't deal with the "Default Lesson" patterns, which look very empty.

These are new patterns that Ronnie shared here: p1654710581646339-slack-C8QJAM52B

> The design post uses large, bold, visible headings, but our patterns do not.

I guess we can do some changes here, but this will probably change based on your theme. Won't it?

> The design post has some "Testimonial" patterns at the top, with nice images, which gives some grounding for the overall design. But I don't think we're actually going to implement these patterns for lessons, right?

Same assumption from my side, I don't think we are implementing those finally. It's also a weird concept a "testimonial" for a lesson.

> There are other differences between the look of the designs in the post, and the actual look of the blocks. For example, the flashcard block in the design does not match the look of the actual flashcard block (in the 1271-gh-Automattic/sensei-pro); the text is more bold, the "Flip card" link has an icon.

As I said I limited myself to bring in the patterns from the demo-site. Let me know if I need to make any adaptions to that. I would say it would be better to have @burtrw or @Luchadores to change the demo-site and I can just bring the changes in (or even in a separate PR).

aaronfc committed:
Merge remote-tracking branch 'origin/add/new-course-patterns' into change/new-lesson-patterns

aaronfc committed:
Use pattern in the lesson template.

aaronfc committed:
Remove quiz question content due to bug.

See #5255

aaronfc commented:
>I'd just ask if you could base it on https://github.com/Automattic/sensei/pull/5246, and update the lesson CPT to use the default pattern, so we avoid the duplicated code to maintain. 😀

@renatho nice suggestion, done in [24be446](https://github.com/Automattic/sensei/pull/5248/commits/24be4460c472d0d4c4c6c1a194d71a2472b7feb2)

Additionally, during testing I realised a [bug](https://github.com/Automattic/sensei/issues/5255) since I added contents to the Quiz in the Default with Quiz pattern so I reverted that part.

aaronfc commented on assets/admin/editor-wizard/steps/course-upgrade-step.js:
```diff
@@ -10,6 +10,7 @@ import { useSelect } from '@wordpress/data';
  */
 import { EXTENSIONS_STORE } from '../../../extensions/store';
 import senseiProUpsellImage from '../../../images/sensei-pro-upsell.png';
+import CheckIcon from '../../../icons/checked.svg';
```
This comes from a change I had from `feature/course-lesson-wizard` that @renatho's parent branch don't have yet. Please ignore all changes in this file.

aaronfc commented on assets/admin/editor-wizard/style.scss:
```diff
@@ -134,15 +134,16 @@
 	&__feature-item {
 		padding: 0 0 20px 24px;
 		margin: 0;
+		display: flex;
```
This comes from a change I had from `feature/course-lesson-wizard` that @renatho's parent branch don't have yet. Please ignore all changes in this file.

fjorgemota replied:
I must say that I was very confused for a moment because there are even commits from myself on this branch. But yeah, fair enough. :P

alexsanford commented:
All of your points sound good to me. I think with the styling changes from my PR we should be ok. It looked cluttered and hard to visually parse, but with the consistent sizing and centering it's not so bad.

So visually, this one is fine by me ✅ 

(I haven't done a review of the code though)

renatho commented on includes/block-patterns/lesson/templates/default-with-quiz.php:
```diff
@@ -0,0 +1,41 @@
+<p></p>
+<!-- /wp:paragraph -->
+
+<!-- wp:sensei-lms/quiz {"options":{"passRequired":false,"quizPassmark":0,"autoGrade":false,"allowRetakes":false,"showQuestions":null,"randomQuestionOrder":false,"failedIndicateIncorrect":true,"failedShowCorrectAnswers":true,"failedShowAnswerFeedback":true,"buttonTextColor":null,"buttonBackgroundColor":null,"pagination":{"paginationNumber":null,"showProgressBar":false,"progressBarRadius":6,"progr ...
```
I think we could use it as:

`<!-- wp:sensei-lms/quiz {"options":{"passRequired":false,"quizPassmark":0,"autoGrade":false,"allowRetakes":false,"showQuestions":null,"randomQuestionOrder":false,"failedIndicateIncorrect":true,"failedShowCorrectAnswers":true,"failedShowAnswerFeedback":true,"buttonTextColor":null,"buttonBackgroundColor":null,"pagination":{"paginationNumber":null,"showProgressBar":false,"progressBarRadius":6,"progressBarHeight":12,"progressBarColor":null,"progressBarBackground":null},"enableQuizTimer":"1","timerValue":300},"isPostTemplate":true} /-->` (the innerblocks are auto added)

I didn't explore why we couldn't use only `<!-- wp:sensei-lms/quiz /-->` (a weird error happens) and get automatically the defaults though.

renatho replied:
Also, I imagine it would make sense to have the quiz as the last thing since it doesn't appear in the lesson. WDYT?

aaronfc replied:
I tried to keep everything "as it is" when copying from the sensei-demo site, to avoid future issues. I am a little concerned that we have to update or add new patterns in the future and we start making inconsistencies between the existing and the new ones...
Anyway I am going to use your suggestion on the quiz block.

Regarding the quiz at the end. I did it like that, but then moved the Lesson Actions block to the end after [this comment](p1654779781166499/1654710581.646339-slack-C8QJAM52B) from Ronnie.
My initial reasoning was the same as yours, but I asked and after Ronnie's comment I changed it to be the Lesson Actions the last element.

renatho reviewed (commented):
In general, it looks good and works well!

I just think we should add the quiz as the last thing in the pattern, since it doesn't appear in the lesson. I also added a suggestion to try to simplify the quiz block code.

aaronfc committed:
Simplify quiz code.

aaronfc committed:
Merge remote-tracking branch 'origin/feature/course-lesson-wizard' into change/new-lesson-patterns

renatho reviewed (approved):
🎉
